Fix Go dependency + Update Brew install Script

alexpeterson91 opened the pull request:
fixes #35 

updated go dependency to `go@1.21` 

updated install script for homebrew on mac

Fixed random formatting errors

alexpeterson91 committed:
update Homebrew install for mac and update go version to match required version by current pocket-core-v0

alexpeterson91 committed:
update wording

alexpeterson91 committed:
add sudo to the new Mac Homebrew install script

alexpeterson91 committed:
set the go dependency to go@1.21 instead of go@1.21.9 which is the latest version in the 1.21 family and whats installed behind the scenes by brew when running brew install go@1.21 but keeping it the same as the Go dependency in pocket-core

alexpeterson91 commented:
@Olshansk

Olshansk commented on README.md:
````diff
@@ -1,39 +1,39 @@
-```bash
-/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```zsh
+sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
````
Why does this need to be sudo?

alexpeterson91 replied:
on most macs this is the output without sudo, 
```
% /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
==> Checking for `sudo` access (which may request your password)...
Password:
```

So i suppose it's not needed in the command but you need the password whether or not you use sudo in the command.
```suggestion
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
I can remove it if you desire.

Olshansk replied:
1. Remove sudoe
2. Revert to bash. I also use zsh but keep it general purpose

alexpeterson91 committed:
remove sudo from brew install command

alexpeterson91 committed:
Update README.md

remove deprecated linuxbrew merge with script for mac homebrew, 
change term coinbase to keybase

alexpeterson91 commented:
@Olshansk got it, will remove sudo for sure, and i was gonna say why not leave the mac homebrew install command with the language defined as zsh because thats just the default shell for all macs and this is a mac command with the linux command separate.  
However i just checked and linuxbrew is deprecated, like 5 years now. was merged into regular homebrew and it's installed the same way so im gonna set the defined language to bash since thats whats needed for linux and mac, and it doesnt affect the way anything is rendered anyway at least in this case. 
Also i inspected [Brew.sh](https://brew.sh) 's home page with the script and it defines the language as bash as well. 
So im gonna make a couple other changes namely removing the linux command and move linux instructions to merge with macOS as they use the exact same command and the exact same source and language.
Let me know if you'd like me to include a link to the homebrew Mac flat package installer `.pkg` file.
and finally i changed the expected output from coinbase passphrase to keybase since thats the terminology used by pocket core, however when i run the start command it first initializes several things then says no keys in keybase, create using X command and exits. let me know if i should update the expected respnse

Olshansk commented:
Please "squash & merge" when ready

alexpeterson91 commented:
> Please "squash & merge" when ready

i dont have write access to merge, must be repo rules

<img width="993" alt="Screenshot 2024-05-02 at 3 44 45 PM" src="https://github.com/pokt-network/homebrew-pocket-core/assets/4649787/39397ade-dded-466e-8d16-62ddb4bc2132">
